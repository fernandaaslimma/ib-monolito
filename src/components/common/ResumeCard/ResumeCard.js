import React, { Fragment } from "react";
import { string } from "prop-types";
import moment from "moment";
import { AlertMessage } from "react-bocombbm-components";
import Icon from "../Icon";
import {
  translate,
  getDateFieldPlaceholderByLocale
} from "../../../utils/i18n";
import { formatCPF, formatCNPJ } from "../../../utils/formatNumber";

import {
  ReceivingInstitutionInfo,
  ReceivingInstitutionName,
  WapperBankImg,
  InfoCard,
  Title,
  SubTitle
} from "./styles";
function ResumeCard({
  cardTitle,
  cardMessage,
  destinyLogoUri,
  destinyName,
  name,
  document,
  destiny,
  purpose,
  creationDate,
  expiration,
  alertMsg,
  ...props
}) {
  const expirationDiff = date => {
    const diff = moment.duration(moment(date).diff(moment(new Date())));
    const meses = diff.asMonths();
    const expirationTime = `${meses.toFixed(0)} ${translate(
      "MONTHS"
    )} | ${moment(date).format(getDateFieldPlaceholderByLocale())}`;

    return expirationTime;
  };

  const originInfo = destinyName && (
    <ReceivingInstitutionInfo>
      {destinyLogoUri === "Bank" ||
      destinyLogoUri === undefined ||
      destinyLogoUri === null ? (
        <Icon type="Bank" width={36} height={36} />
      ) : (
        <WapperBankImg src={destinyLogoUri} />
      )}
      <ReceivingInstitutionName>{destinyName}</ReceivingInstitutionName>
    </ReceivingInstitutionInfo>
  );

  return (
    <Fragment>
      {cardTitle && originInfo && (
        <InfoCard data-test={props.dataTest}>
          <Title>{cardTitle}</Title>
          {cardMessage}
          {originInfo}
        </InfoCard>
      )}
      <InfoCard>
        <Title>{translate("OPEN_BANKING_SHARING_SUMMARY")}</Title>
        {name && (
          <Fragment>
            {translate("OPEN_BANKING_NAME")}
            <SubTitle data-test="nameLabel">{name}</SubTitle>
          </Fragment>
        )}
        {document && (
          <Fragment>
            {document.length === 11
              ? translate("OPEN_BANKING_CPF")
              : document.length === 14
              ? translate("OPEN_BANKING_CNPJ")
              : translate("OPEN_BANKING_DOCUMENT")}
            <SubTitle data-test="cpfOrCnpjLabel">
              {document.length === 11
                ? formatCPF(document)
                : document.length === 14
                ? formatCNPJ(document)
                : document}
            </SubTitle>
          </Fragment>
        )}
        {destiny && (
          <Fragment>
            {translate("OPEN_BANKING_DESTINATION_INSTITUTION")}
            <SubTitle data-test="destinationInstitutionLabel">
              {destiny}
            </SubTitle>
          </Fragment>
        )}
        {purpose && (
          <Fragment>
            {translate("OPEN_BANKING_PURPOSE_DATA")}
            <SubTitle data-test="consentPurpose">{purpose}</SubTitle>
          </Fragment>
        )}

        {creationDate && (
          <Fragment>
            {translate("OPEN_BANKING_CONFIRMATION_DATA")}
            <SubTitle data-test="creationDate">{creationDate}</SubTitle>
          </Fragment>
        )}
        {expiration && (
          <Fragment>
            {translate("OPEN_BANKING_SHARE_TERM")}
            <SubTitle data-test="expirationDateTime">
              {expirationDiff(expiration)}
            </SubTitle>
          </Fragment>
        )}

        {alertMsg && (
          <AlertMessage
            icon="Attention"
            type="neutral"
            spacing={{
              top: "s",
              bottom: "none",
              right: "none",
              left: "none"
            }}
          >
            {alertMsg}
          </AlertMessage>
        )}
      </InfoCard>
    </Fragment>
  );
}

ResumeCard.defaultProps = {
  cardTitle: null,
  cardMessage: null,
  destinyLogoUri: null,
  destinyName: null,
  name: null,
  document: null,
  destiny: null,
  purpose: null,
  expiration: null,
  dataTest: undefined,
  alertMsg: null
};

ResumeCard.propTypes = {
  cardTitle: string,
  cardMessage: string,
  destinyLogoUri: string,
  destinyName: string,
  name: string,
  document: string,
  destiny: string,
  purpose: string,
  expiration: string,
  dataTest: string,
  alertMsg: string
};

export default ResumeCard;
