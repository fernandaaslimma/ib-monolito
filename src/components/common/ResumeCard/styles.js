import styled, { css } from "styled-components";
import { white, gray200, gray300, lighestgrey } from "../../../styles/settings";
import { rem, remFontSize, media } from "../../../styles/tools/index";

export const Card = styled.section`
  background: ${white};
  border: 1px solid #eef1f3;
  box-sizing: border-box;
  border-radius: 8px;
`;

export const RedirectionInfo = styled.div`
  height: 64px;
  background: #eef1f3;
  display: flex;
  align-items: center;
  padding: 0 ${rem(40)} 0 ${rem(40)};
`;

export const RedirectionText = styled.span`
  font-family: Lato;
  font-style: normal;
  font-weight: bold;
  font-size: 12px;
  line-height: 16px;
  padding: 0 0 0 ${rem(18)};
  /* or 133% */

  color: #003b78;
`;

export const ReceivingInstitutionInfo = styled.div`
  display: flex;
  align-items: center;
  margin: ${rem(16)} ${rem(24)} ${rem(16)} 0;
`;

export const ReceivingInstitutionName = styled.span`
  font-family: Lato;
  font-style: normal;
  font-weight: bold;
  font-size: 16px;
  line-height: 24px;
  /* identical to box height, or 150% */

  display: flex;
  align-items: center;

  color: #2d4758;

  margin-left: ${rem(16)};
`;

export const IconContainer = styled.span`
  align-items: center;
  background: ${white};
  border-radius: ${rem(4)};
  color: ${lighestgrey};
  display: inline-flex;
  height: ${rem(24)};
  justify-content: center;
  margin-right: ${rem(9)};
  width: ${rem(24)};

  ${({ iconContainerBg }) =>
    iconContainerBg &&
    css`
      background: rgba(${iconContainerBg}, 0.2);
    `};

  ${media.md(css`
    height: ${rem(28)};
    width: ${rem(28)};
    margin-right: ${rem(13)};
  `)};
`;

export const Header = styled.header`
  align-items: center;
  display: flex;
`;

export const Section = styled.section`
  background-color: ${white};
  border-radius: ${rem(2)};
  padding: ${rem(12)};
  box-shadow: 0 ${rem(1)} ${rem(12)} ${rem(6)} rgba(211, 225, 232, 0.13);
  transition: height 0.8s linear;
  @media print {
    box-shadow: none;
    border-radius: none;
  }
  ${({ height }) =>
    height &&
    css`
      height: ${rem(height)};
    `};
  ${({ noBg }) =>
    noBg &&
    css`
      background: transparent;
    `};
  ${media.md(css`
    padding: ${rem(20)};
  `)};

  ${({ hasChildren }) =>
    hasChildren &&
    css`
      ${Header} {
        margin-bottom: ${rem(20)};

        ${media.md(css`
          margin-bottom: ${rem(20)};
        `)};

        ${media.lg(css`
          margin-bottom: ${rem(10)};
        `)};
      }
    `};

  ${({ styles }) =>
    styles &&
    css`
      ${styles};
    `};
`;

export const WapperBankImg = styled.img`
  display: flex;
  align-items: center;
  justify-content: center;
  height: ${rem(36)};
  width: ${rem(36)};
  border-radius: ${rem(6)};
  border: 1px solid #d9e0e4;
`;

export const InfoCard = styled.div`
  padding: ${rem(16)} ${rem(15)} ${rem(16)} ${rem(15)};
  background: ${white};
  overflow: auto;
  font-family: Lato;
  font-size: ${remFontSize(14)};
  color: ${gray200};
  line-height: ${remFontSize(22)};
  margin-bottom: ${rem(16)};
`;

export const Title = styled.h2`
  font-size: ${remFontSize(18)};
  margin: 0 0 ${rem(15)} 0;
  color: ${gray300};
  font-weight: 700;
  line-height: 125%;
  /* identical to box height, or 20px */

  letter-spacing: 0.452308px;
`;

export const SubTitle = styled.h2`
  font-size: ${remFontSize(16)};
  margin: 0 0 ${rem(15)} 0;
  color: ${gray300};
  font-weight: 400;
  line-height: ${rem(24)};
  /* identical to box height, or 20px */

  letter-spacing: 0.452308px;
`;
