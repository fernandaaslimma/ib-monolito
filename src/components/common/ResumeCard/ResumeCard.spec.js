import React from "react";
import { shallow } from "enzyme";

import ResumeCard from "./ResumeCard";

describe("ResumeCard component", () => {
  it("should match snapshot with empty content", () => {
    const wrapper = shallow(<ResumeCard />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should match snapshot with BBM resume", () => {
    const wrapper = shallow(
      <ResumeCard
        dataTest="resumeCard"
        cardTitle="Titulo"
        cardMessage="Texto explicativo"
        destinyLogoUri="Bank"
        destinyName="BBM"
        name="Fulano"
        document="09987594477"
        destiny="BBM"
        purpose="Motivo do Consentimento"
        expiration="1641820886000"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
