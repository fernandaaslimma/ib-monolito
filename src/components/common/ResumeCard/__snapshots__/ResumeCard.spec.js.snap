// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResumeCard component should match snapshot with BBM resume 1`] = `
<Fragment>
  <styled.div
    data-test="resumeCard"
  >
    <styled.h2>
      Titulo
    </styled.h2>
    Texto explicativo
    <styled.div>
      <Icon
        color={null}
        cursor={false}
        fixCrossAxisAlign={null}
        height={36}
        type="Bank"
        width={36}
      />
      <styled.span>
        BBM
      </styled.span>
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.h2 />
    <styled.h2
      data-test="nameLabel"
    >
      Fulano
    </styled.h2>
    <styled.h2
      data-test="cpfOrCnpjLabel"
    >
      099.875.944-77
    </styled.h2>
    <styled.h2
      data-test="destinationInstitutionLabel"
    >
      BBM
    </styled.h2>
    <styled.h2
      data-test="consentPurpose"
    >
      Motivo do Consentimento
    </styled.h2>
    <styled.h2
      data-test="expirationDateTime"
    >
      0 undefined | 06/01/2010
    </styled.h2>
  </styled.div>
</Fragment>
`;

exports[`ResumeCard component should match snapshot with empty content 1`] = `
<Fragment>
  <styled.div>
    <styled.h2 />
  </styled.div>
</Fragment>
`;
